crates/storage/db/src/implementation/mdbx/tx.rs

```diff
@@ -28,12 +33,18 @@ impl<'env, K: TransactionKind, E: EnvironmentKind> Tx<'env, K, E> {
     where
         'a: 'env,
     {
-        Self { inner, db_handles: Default::default() }
+        Self { inner, db_handles: Default::default(), metrics_handler: None }
+    }
+
+    /// Sets the [MetricEventsSender] to report metrics about the transaction and cursors.
+    pub fn with_metrics_tx(mut self, metrics_tx: MetricEventsSender) -> Self {
+        self.metrics_handler = Some(MetricsHandler { txn_id: self.id(), metrics_tx });
+        self
     }
 
     /// Gets this transaction ID.
     pub fn id(&self) -> u64 {
-        self.inner.id()
+        self.metrics_handler.as_ref().map_or_else(|| self.inner.id(), |handler| handler.txn_id)
```

Comment:
where are we using this function

this change seems unecessary?

Reply:
`self.inner.id()` makes a FFI each time it's called, but for metrics purposes we cache the ID inside `MetricsHandler`. If the database is used with metrics enabled, there's no need to make an FFI to get the ID because we already have it.

We don't use this method anywhere beyond metrics in our codebase, but it's public, so this change is a small perf optimization.